### PR TITLE
added store.fs.rmdir in storage.py to eliminate slow rmdir operation

### PR DESCRIPTION
**Pull request by hayesgb**
This is a fix to zarr/storage.py that add the fs attribute to store.

TODO:
* [ ] Add unit tests and/or doctests in docstrings
* [ ] Add docstrings and API docs for any new/modified user-facing classes and functions
* [ ] New/modified features documented in docs/tutorial.rst
* [ ] Changes documented in docs/release.rst
* [ ] AppVeyor and Travis CI passes
* [ ] Test coverage is 100% (Coveralls passes)

### CODE REVIEW

File: zarr/storage.py [hunk 1]
```diff
@@ -119,6 +119,8 @@ def rmdir(store, path: Path = None):
     if hasattr(store, 'rmdir'):
         # pass through
         store.rmdir(path)
+    elif hasattr(store.fs, 'rmdir'):
```
Comment (Carreau) [inline zarr/storage.py:122]:
```suggestion
    elif hasattr(store, 'fs') and hasattr(store.fs, 'rmdir'):
```

**Comment by Carreau** [comment]
I pushed a commit that should fix the first test failures; I believe that should also fix the subsequent failures as well.

**Comment by joshmoore** [comment]
@hayesgb : do you mind adding a test that makes use of `self.store.fs`?

**Comment by hayesgb** [comment]
Sorry for the delay in responding.  No, I don't mind, but it may be a bit until I can get to it.  I added this to eliminate the fallback to a slow implementation of the delete

**Comment by jhamman** [comment]
I'm going to close this as stale. Folks should feel free to reopen if there is interest in continuing this work.

**Pull request closed without merging**